### ISSUE TITLE
Duplicate page requests are made when the PageController contains initial data

### ISSUE THREAD

**Issue by hacker1024** [EdsonBueno/infinite_scroll_pagination#12]
I'm using MobX to store data, and I cache the first page of it on startup. To load the pagination widget with initial data, I have this in my initState:

```dart
_pagingController.value = PagingState(
  nextPageKey: _store.nextPageOffset,
  itemList: _store.loadedItems,
  error: _store.errorMessage,
);
```

The problem is that the widget always requests the page at `nextPageKey` in its `initState()`, which we can see[ here](https://github.com/EdsonBueno/infinite_scroll_pagination/blob/b40b3ea44bf0161d17b4becaae645f2ae3feb580/lib/src/ui/paged_sliver_builder.dart#L148).

This is an erranous duplicate request, as since there's data that's already loaded, [the list item builder will also request more data ](https://github.com/EdsonBueno/infinite_scroll_pagination/blob/b40b3ea44bf0161d17b4becaae645f2ae3feb580/lib/src/ui/paged_sliver_builder.dart#L228)(with the same key, but a different `_fetchTriggerIndex`).

I believe this can be fixed by simply checking if there's existing data in initState, and skipping the request accordingly.

> I originally incorrectly identified the cause of this issue, oops! The old (incorrect) report is below.
>
> ~~I'm using MobX for state management, and I pre-download the first page of items when my app starts up.
> As such, I provide a non-zero integer to the `firstPageKey` in my `PaginController`.~~
> 
> ~~The problem is that regardless of this value, `_lastFetchTriggerIndex` is always set to zero when the first page loads, as we can see [here](https://github.com/EdsonBueno/infinite_scroll_pagination/blob/b40b3ea44bf0161d17b4becaae645f2ae3feb580/lib/src/ui/paged_sliver_builder.dart#L148).~~
> 
> ~~As the page started with data, items are already being built. As a result of this, a page load is requested again, [here](https://github.com/EdsonBueno/infinite_scroll_pagination/blob/b40b3ea44bf0161d17b4becaae645f2ae3feb580/lib/src/ui/paged_sliver_builder.dart#L228).~~
> 
> ~~This would normally be skipped, because `_lastFetchTriggerIndex` should be equal to the `newFetchTriggerIndex`. It's not, however, because it's zero instead of the `firstPageKey` initially.~~
> 
> ~~This results in the page being requested twice.~~

**Comment by hacker1024** [comment]
A non-zero `firstPageKey` also doesn't play nice with refreshing.

Perhaps there should be an argument in the controller to provide initial data? That way `firstPageKey` can only be used to represent the key for the first page of data.

**Comment by EdsonBueno** [comment]
Thanks @hacker1024 for diagnosing the issue and opening the PR!
Your fix is included in today's release (2.2.2) and you're officially one of the package's contributors!
